source/dem/particle_wall_linear_force.cc

```diff
@@ -214,6 +215,187 @@ ParticleWallLinearForce<dim>::calculate_particle_wall_contact_force(
   this->mpi_correction_over_calculation_of_forces_and_torques();
 }
 
+
+template <int dim>
+void
+  ParticleWallLinearForce<dim>::calculate_particle_floating_wall_contact_force(
+    std::vector<
+      std::map<typename Triangulation<dim - 1, dim>::active_cell_iterator,
+               std::unordered_map<types::particle_index,
+                                  particle_wall_contact_info_struct<dim>>,
+               dem_data_containers::cut_cell_comparison<dim>>>
+      &                        particle_floating_mesh_in_contact,
+    const double &             dt,
+    std::vector<Tensor<1, 3>> &torque,
+    std::vector<Tensor<1, 3>> &force,
+    const std::map<unsigned int, Tensor<1, 3>>
+      &floating_mesh_translational_velocity,
+    const std::map<unsigned int, Tensor<1, 3>>
+      &                                     floating_mesh_rotational_velocity,
+    const std::map<unsigned int, Point<3>> &floating_mesh_center_of_rotation)
```

Comment:
Wouldnt it have been simpler to store the map of active cell iterator INTO the rigid solid object and so you could just pass a rigid solid object instead? It would be a lot easier to understand and it would prevent all of these auxiliary containers for the velocity. Why not keep the better encapsulation?

Reply:
Applied.